### PR TITLE
Allowing ability to observe partial state.

### PR DESCRIPTION
**Pull request by mhusainisurge**
This is to implement enhancement #4. I generalized it a bit to allow using any arbitrary selector on the state. This is the my first contribution on Github, so I hope I got everything right. 😃

### CODE REVIEW

File: ReduxSimple/ReduxStore.cs [hunk 1]
```diff
@@ -34,10 +34,16 @@ public IObservable<TState> ObserveState()
...
+        public IObservable<TPartial> ObserveState<TPartial>(Func<TState, TPartial> selector)
+        {
+            return _stateSubject.Select(selector).DistinctUntilChanged();
+        }
+
         public IObservable<object> ObserveAction()
         {
             return _actionSubject.AsObservable();
         }
+
```
Comment (Odonno) [inline ReduxSimple/ReduxStore.cs]: Overloaded methods (with the same name) should be close to one another. The same should be applied for `ObserveState`.

**Comment by Odonno** [comment]
@mhusainisurge Ey. Thanks for your first time contribution! I really enjoy seeing someone with interest on this project.

Your code looks good. Will need to update the docs in consequence, it is no a subject on this PR so I don't force you with it. Anyway, if you make the changes on the line return on the file, I'll consider this PR OK and merged it right after.

On the meantime, will figure out why appveyor builds are failing..

**Comment by Odonno** [comment]
Ok, fixed. Build is now working like expected.

File: ReduxSimple/ReduxStore.cs (hunk 1 above)
Comment (mhusainisurge) [reply ReduxSimple/ReduxStore.cs]: Will do.

**Comment by mhusainisurge** [comment]
Thanks for considering my PR. I have implemented the changes requested. Looking forward to more contributions!

**Pull request merged**